Eventbrite/EventbriteSync.ascx.cs

```diff
@@ -0,0 +1,177 @@
+            {
+                string groupId = PageParameter( "GroupId" );
+                if ( !string.IsNullOrWhiteSpace( groupId ) && rocks.kfs.Eventbrite.Eventbrite.EBUsageCheck( groupId.AsInteger() ) )
+                {
+                    ShowPanels( groupId.AsInteger() );
+                }
+                else
+                {
+                    pnlEventbriteButtons.Visible = false;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Handles the BlockUpdated event of the control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void Block_BlockUpdated( object sender, EventArgs e )
+        {
+            string groupId = PageParameter( "GroupId" );
+            if ( !string.IsNullOrWhiteSpace( groupId ) && rocks.kfs.Eventbrite.Eventbrite.EBUsageCheck( groupId.AsInteger() ) )
+            {
+                ShowPanels( groupId.AsInteger() );
+            }
+            else
+            {
+                pnlEventbriteButtons.Visible = false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the group.
+        /// </summary>
+        /// <param name="groupId">The group identifier.</param>
+        /// <returns></returns>
+        private Group GetGroup( int groupId, RockContext rockContext = null )
+        {
+            string key = string.Format( "Group:{0}", groupId );
+            Group group = RockPage.GetSharedItem( key ) as Group;
+            if ( group == null )
+            {
+                rockContext = rockContext ?? new RockContext();
+                group = new GroupService( rockContext ).GetNoTracking( groupId );
+                RockPage.SaveSharedItem( key, group );
+            }
+
+            return group;
+        }
+
+        private void ShowPanels( int groupId )
+        {
+            Group group = null;
+
+            RockContext rockContext = new RockContext();
+
+            if ( !groupId.Equals( 0 ) )
+            {
+                group = GetGroup( groupId, rockContext );
+
+                pnlEventbriteButtons.Visible = IsUserAuthorized( Authorization.EDIT ) || group.IsAuthorized( Authorization.EDIT, this.CurrentPerson );
+
+                hfGroupId.Value = group.Id.ToString();
+            }
+        }
+
+        /// <summary>
+        /// Handles the Click event of the lbSyncButton control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void lbSyncButton_Click( object sender, EventArgs e )
+        {
+            int groupId = hfGroupId.ValueAsInt();
+            var enableLogging = GetAttributeValue( "EnableLogging" ).AsBoolean();
+            Server.ScriptTimeout = Server.ScriptTimeout * 2;
```

Comment:
See if we can do this incrementally

Reply:
Attempting to do it in assembly. Will need to test it on a clients server.